k8s_helm_charts/zdm/templates/deployment.yaml

```diff
@@ -0,0 +1,110 @@
+# pull in these templates outside of range loop
+{{ $zdm_fullname := include "zdm.fullname" . -}}
+{{- $zdm_labels := include "zdm.labels" . -}}
+{{- $zdm_selectorLabels := include "zdm.selectorLabels" . -}}
+
+# calculate a variable that contains all proxy service addresses
+{{ $service_addresses := "" -}}
+{{- range $index := until (.Values.proxy.count | int) -}}
+    {{- $service_addresses = printf "%s,$(%s_%s_SERVICE_HOST)" $service_addresses ($zdm_fullname | upper | replace "-" "_") ($index | toString) -}}
+{{- end -}}
+{{- $service_addresses = $service_addresses | trimPrefix "," -}}
+
+{{- range $index := until (.Values.proxy.count | int) -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    {{- $zdm_labels | nindent 4 }}
+    app: {{ $zdm_fullname }}-{{ $index }}
+  name: {{ $zdm_fullname }}-{{ $index }}
+  namespace: {{ $.Values.namespace }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- $zdm_selectorLabels | nindent 6 }}
+      app: {{ $zdm_fullname }}-{{ $index }}
+  template:
+    metadata:
+      labels:
+        {{- $zdm_selectorLabels | nindent 8 }}
+        app: {{ $zdm_fullname }}-{{ $index }}
+    spec:
+      containers:
+      - image: "{{ $.Values.proxy.image.repository }}:{{ $.Values.proxy.image.tag | default $.Chart.AppVersion }}"
+        name: {{ $zdm_fullname }}-{{ $index }}
+        resources:
+          requests:
+            memory: {{ $.Values.proxy.resources.requests.memory | quote }}
+            cpu: {{ $.Values.proxy.resources.requests.cpu | quote }}
+          limits:
+            memory: {{ $.Values.proxy.resources.limits.memory | quote }}
+            cpu: {{ $.Values.proxy.resources.limits.cpu | quote }}
+        envFrom:
+        - configMapRef:
+            name: {{ $zdm_fullname }}
+        env:
+        - name: ZDM_PROXY_LISTEN_ADDRESS
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        - name: ZDM_PROXY_LISTEN_PORT
+          value: "9042"
+        - name: ZDM_METRICS_ADDRESS
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        - name: ZDM_METRICS_PORT
+          value: "14001"
+        - name: ZDM_PROXY_TOPOLOGY_INDEX
+          value: {{ $index | quote }}
+        - name: ZDM_PROXY_TOPOLOGY_ADDRESSES
+          value: {{ $service_addresses }}
```

Comment:
Are `service_addresses` actual IP address or DNS names?
I remember I discussed this with @joao-r-reis some time ago and since these are used to spoof the peers addresses in the `system.peers` table and the data type is `inet`, it could cause problems to have it as DNS names because of the incompatible type.
At least that was the case back in February when I added the ZDM proxy in a Helm chart, I'm not sure if things changed since then.

Reply:
Those environmental variables are exposed as IP addresses. They're a built-in support from k8s and all pods have access to these environmental variables to get to the IP addresses of all running services. See this article for more details: https://learn.microsoft.com/en-us/visualstudio/bridge/kubernetes-environment-variables

Reply:
Ok, cool. Yeah, I'm familiar with that, I just didn't notice the env var **value** was being extracted with the `$(...)` notation, I thought it was using the var name as the output. That looks good then 👍